app/src/ereferrals/java/org/eyeseetea/malariacare/strategies/DashboardActivityStrategy.java

```diff
@@ -227,12 +233,64 @@ private void openNewSurvey(Activity activity) {
         // Put new survey in session
         SurveyDB survey = new SurveyDB(null, programDB, Session.getUserDB());
         survey.save();
+        saveValuesFromIntent(activity, survey);
         Session.setMalariaSurveyDB(survey);
         //Look for coordinates
         prepareLocationListener(activity, survey);
         mDashboardActivity.initSurvey();
     }
 
+    private void saveValuesFromIntent(Activity activity, SurveyDB survey) {
+        if (activity.getIntent() != null && activity.getIntent().getExtras() != null){
+            String values = activity.getIntent().getExtras().getString(LoginActivityStrategy.VALUES_BUNDLE);
+            if(values !=null && !values.isEmpty()){
+                try {
+                    survey.save();
+                    parseJsonValuesAndSave(survey, values);
+                    if(survey.getValuesFromDB().size()>0) {
+                        DashboardActivity.dashboardActivity.setPreLoadSurveyOpenning(true);
+                    }
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+            activity.getIntent().removeExtra(LoginActivityStrategy.VALUES_BUNDLE);
+            activity.getIntent().removeExtra(LoginActivityStrategy.CREATED_SURVEY_FROM_OTHER_APP);
+        }
+    }
+
+    private void parseJsonValuesAndSave(SurveyDB survey, String jsonValues) throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
+        ConnectVoucher connectVoucher;
+            connectVoucher = mapper.readValue(jsonValues, ConnectVoucher.class);
+        Iterator it = connectVoucher.getValues().entrySet().iterator();
+        while (it.hasNext()) {
+            Map.Entry pair = (Map.Entry)it.next();
+            saveValue(survey, pair.getKey().toString(),  pair.getValue().toString());
+            it.remove(); // avoids a ConcurrentModificationException
+        }
+    }
+
+    private void saveValue(SurveyDB survey, String key, String value) {
```

Comment:
this method seems duplicate with others on the app, can you refactor to avoid duplicate code?